app/src/androidTest/java/org/simple/clinic/reports/ReportsSyncAndroidTest.kt

```diff
@@ -0,0 +1,65 @@
+package org.simple.clinic.reports
+
+import androidx.test.runner.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import org.simple.clinic.AuthenticationRule
+import org.simple.clinic.TestClinicApp
+import org.simple.clinic.util.Just
+import org.simple.clinic.util.None
+import org.simple.clinic.util.RxErrorsRule
+import java.io.File
+import javax.inject.Inject
+
+@RunWith(AndroidJUnit4::class)
+class ReportsSyncAndroidTest {
+
+  @Inject
+  lateinit var reportsRepository: ReportsRepository
+
+  @Inject
+  lateinit var reportsSync: ReportsSync
+
+  private val authenticationRule = AuthenticationRule()
+
+  private val rxErrorsRule = RxErrorsRule()
+
+  @get:Rule
+  val ruleChain = RuleChain
+      .outerRule(authenticationRule)
+      .around(rxErrorsRule)!!
+
+  @Before
+  fun setUp() {
+    TestClinicApp.appComponent().inject(this)
+  }
+
+  @After
+  fun tearDown() {
+    reportsRepository
+        .reportsFile()
+        .blockingFirst()
+        .toNullable()
+        ?.let { it.delete() }
+  }
+
+  @Test
+  fun when_pulling_reports_from_the_server_it_should_save_the_reports_as_a_file() {
+    val reportsFileBeforeSync = reportsRepository.reportsFile().blockingFirst()
+
+    reportsSync
+        .pull()
+        .test()
+        .assertNoErrors()
+
+    val reportsFileAfterSync = reportsRepository.reportsFile().blockingFirst() as Just<File>
```

Comment:
Not important, but you can use `unwrapJust()` here.

```kotlin
reportsRepository.reportsFile()
  .unwrapJust()
  .blockingFirst()
```

Reply:
Better. I'll do this.